DOC-1772: Added doc for text_patterns_lookup (#2517)
tinymce/tinymce-docs: modules/ROOT/partials/configuration/text_patterns_lookup.adoc

[[text_patterns_lookup]]
== `+text_patterns_lookup+`

> This option allows specifying a function that dynamically appends additional text patterns to the xref:content-behavior-options.adoc#text_patterns[text_patterns]. The function takes a context object as an argument and should return a list of text patterns. The function will be called whenever the editor is attempting to do text pattern matching.

Reviewer: ```suggestion
This option allows the specification of a function that dynamically appends additional text patterns to the default xref:content-behavior-options.adoc#text_patterns[text_patterns]. The function takes a context object as an argument and returns an array of text patterns. This function is called whenever the editor triggers text pattern matching.
```

*Type:* `+Function+`

=== Example: Using text_patterns_lookup

[source,js]
----
tinymce.init({
  selector: 'textarea', // change this value according to your HTML
  text_patterns_lookup: (ctx) => {
    const parentTag = ctx.block.nodeName.toLowerCase();
    if (parentTag === 'pre') {
      return [
        { start: '`', end: '`', format: 'code' }
      ];
    } else if (parentTag === 'p') {
      return [
        { start: '*', end: '*', format: 'bold' }
      ];
    } else {
      return [
        // ctx.text is the string from the start of the block to the cursor
        { start: 'brb', replacement: ctx.text + ': Be Right Back' }
      ];

>     }

Reviewer: Should this also be an `else if`? Seems like returning this in all other cases isn't a great idea. Maybe just returning a list of text patterns when within a `p` tag is adequate? Just from what I can tell, but I don't really know the option.

Reviewer 2: Maybe just have the last else be `return [];` showing how you should return things that doesn't match what you are looking for.

[[text_patterns_lookup]]
== `+text_patterns_lookup+`

> This option allows you to specify a function that dynamically append additional text patterns to the xref:content-behavior-options.adoc#text_patterns[text_patterns]. The function takes a context object as an argument and should return a list of text patterns. The function will be called whenever the editor is attempting to do text pattern matching.

Reviewer 3: No `you`, you shouldn't use pronouns per https://github.com/tinymce/tinymce-docs/blob/staging/docs-6/CONTRIBUTING.md#pronouns

*Type:* `+Function+`

=== Example: Using text_patterns_lookup

[source,js]
----
tinymce.init({
  selector: 'textarea', // change this value according to your HTML
  text_patterns_lookup: (ctx) => {

>     const text = ctx.

Reviewer 3: This isn't valid or is missing something (probably the latter).
